torch/_library/autograd.py

```diff
@@ -1,26 +1,39 @@
-from typing import Any, Callable, Optional
+import dataclasses
+from typing import Any, Callable, Optional, Protocol
 
 from .. import _C, autograd, Tensor
 
 from ..utils import _pytree
 from . import utils
 
 
-def make_autograd_impl(opdef: Any) -> Callable:
-    name: str = f"GeneratedBackwardFor_{opdef._namespace}_{opdef._name}"
+class InfoProtocol(Protocol):
+    _setup_context_fn: Optional[Callable]
+    _backward_fn: Optional[Callable]
+
+
+@dataclasses.dataclass
+class Info:
+    _setup_context_fn: Optional[Callable]
+    _backward_fn: Optional[Callable]
+
+
+# op actually has type "torch._ops.OpOverload", but we've got some circular import issues.
+def make_autograd_impl(op: Any, info: InfoProtocol) -> Callable:
```

Comment:
```suggestion
def make_autograd_impl(op: "torch._ops.OpOverload", info: InfoProtocol) -> Callable:
```

That's why strings as types are allowed!

Reply:
mypy is smarter nowadays; it will complain that torch isn't imported if we do "torch._ops.OpOverload" :/

Reply:
Ho and it will complain that torch is not used as well?
We can have an `import torch` at the top of the file right?

Reply:
I got away with `from . import _ops`, but `import torch` at the top could trigger circular import issues